### PR TITLE
Rewrite container_stats tests

### PR DESCRIPTION
**Pull request by apostasie**
On top of #3617

### CODE REVIEW

File: cmd/nerdctl/container/container_stats_test.go [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+				return helpers.Command("container", "stats", "--no-stream", data.Get("id"))
+			},
+			Expected: test.Expects(0, nil, nil),
+		},
+		{
+			Description: "no mem limit set",
+			Command: func(data test.Data, helpers test.Helpers) test.TestableCommand {
+				return helpers.Command("stats", "--no-stream")
+			},
+			// FIXME: "16EiB". why? what is this testing?
```
Comment (AkihiroSuda) [inline cmd/nerdctl/container/container_stats_test.go:94]: There was a bug that printed `UINT64_MAX` as the memory limit

Comment (apostasie) [reply cmd/nerdctl/container/container_stats_test.go:94]: I ll add a comment

Comment (apostasie) [reply cmd/nerdctl/container/container_stats_test.go]: Done.

**Comment by apostasie** [comment]
Failure is part of the legacy/flaky tests (usual `TestComposeUp`).

Can you poke the CI for a retry?

**Review by AkihiroSuda — APPROVED** [review]
Thanks

**Pull request merged**